analyzers/src/SonarAnalyzer.Common/Rules/Utilities/AnalysisWarningAnalyzerBase.cs

```diff
@@ -0,0 +1,61 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.IO;
+using SonarAnalyzer.CFG.Helpers;
+
+namespace SonarAnalyzer.Rules;
+
+public abstract class AnalysisWarningAnalyzerBase : UtilityAnalyzerBase
+{
+    private const string DiagnosticId = "S9999-warning";
+    private const string Title = "Analysis Warning generator";
+
+    private static readonly object FileWriteLock = new();
+
+    protected virtual int MinimalSupportedRoslynVersion { get; } = RoslynHelper.MinimalSupportedMajorVersion;   // For testing
+
+    protected AnalysisWarningAnalyzerBase() : base(DiagnosticId, Title) { }
+
+    protected sealed override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterCompilationAction(c =>
+            {
+                ReadParameters(c);
+                if (IsAnalyzerEnabled && !RoslynHelper.IsRoslynCfgSupported(MinimalSupportedRoslynVersion))     // MsBuild 15 is bound with Roslyn 2.x, where Roslyn CFG is not available.
+                {
+                    // This can be removed after we bump Microsoft.CodeAnalysis references to 3.0 or higher.
```

Comment:
Bumping the version should have a full specification because there will be a lot of stuff that can be removed. This should be easy to find, as `RoslynHelper` should go away as part of that.

I will not add an issue as we won't work on it for a couple of years.